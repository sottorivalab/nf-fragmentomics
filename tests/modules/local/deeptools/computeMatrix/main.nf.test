nextflow_process {

    name "Test Process COMPUTEMATRIX"
    script "modules/local/deeptools/computeMatrix/main.nf"
    process "COMPUTEMATRIX"

    test("Should run COMPUTEMATRIX without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = [
                    [ caseid:'test', sampleid: 'test', timepoint: 'T1' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bigwig/test.bigwig', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/baits.bed', checkIfExists: true),
                    'test_source',
                    ['test_target'],
                    [file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true)]
                ]
                """
            }
        }

        then {
            assert process.success
        }

    }

}
