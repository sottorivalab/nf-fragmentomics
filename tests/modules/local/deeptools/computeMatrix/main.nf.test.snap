{
    "Should run COMPUTEMATRIX without failures": {
        "content": [
            [
                [
                    {
                        "caseid": "test",
                        "sampleid": "test",
                        "timepoint": "T1"
                    },
                    {
                        "source": "test",
                        "name": "test"
                    },
                    "test_matrix.gz:md5,dcf3ebf50243448dddc584a60057a0c1"
                ]
            ],
            [
                "versions.yml:md5,defce74e73df4e345c6c325220d4072d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T14:32:17.347144"
    }
}